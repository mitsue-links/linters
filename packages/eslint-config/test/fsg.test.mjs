import path from 'path';
import {lintFile} from './helper.mjs';

test('fsg_ok.mjs', async () => {
    const filePath = 'fsg_ok.mjs';
    const data = await lintFile(path.join('fixtures', filePath));

    expect(data).toStrictEqual([{
        filePath,
        messages: [],
    }]);
});

test('fsg_error.mjs', async () => {
    const filePath = 'fsg_error.mjs';
    const data = await lintFile(path.join('fixtures', filePath));

    expect(data).toMatchSnapshot();
});
