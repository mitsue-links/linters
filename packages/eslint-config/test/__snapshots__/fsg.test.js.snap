// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fsg_error.js 1`] = `
Array [
  Object {
    "filePath": "fsg_error.js",
    "messages": Array [
      Object {
        "line": 3,
        "ruleId": "no-implicit-globals",
        "severity": 1,
      },
      Object {
        "line": 3,
        "ruleId": "no-undef",
        "severity": 2,
      },
      Object {
        "line": 6,
        "ruleId": "no-implicit-globals",
        "severity": 1,
      },
      Object {
        "line": 6,
        "ruleId": "no-undef",
        "severity": 2,
      },
      Object {
        "line": 10,
        "ruleId": "one-var",
        "severity": 2,
      },
      Object {
        "line": 11,
        "ruleId": "indent",
        "severity": 2,
      },
      Object {
        "line": 12,
        "ruleId": "indent",
        "severity": 2,
      },
      Object {
        "line": 13,
        "ruleId": "indent",
        "severity": 2,
      },
      Object {
        "line": 13,
        "ruleId": "no-implicit-globals",
        "severity": 1,
      },
      Object {
        "line": 13,
        "ruleId": "no-undef",
        "severity": 2,
      },
      Object {
        "line": 13,
        "ruleId": "no-unused-expressions",
        "severity": 2,
      },
      Object {
        "line": 13,
        "ruleId": "no-sequences",
        "severity": 2,
      },
      Object {
        "line": 14,
        "ruleId": "indent",
        "severity": 2,
      },
      Object {
        "line": 14,
        "ruleId": "no-undef",
        "severity": 2,
      },
      Object {
        "line": 14,
        "ruleId": "no-undef",
        "severity": 2,
      },
      Object {
        "line": 23,
        "ruleId": "brace-style",
        "severity": 2,
      },
      Object {
        "line": 23,
        "ruleId": "brace-style",
        "severity": 2,
      },
      Object {
        "line": 32,
        "ruleId": "camelcase",
        "severity": 2,
      },
      Object {
        "line": 46,
        "ruleId": "radix",
        "severity": 2,
      },
      Object {
        "line": 47,
        "ruleId": "radix",
        "severity": 2,
      },
      Object {
        "line": 55,
        "ruleId": "no-eval",
        "severity": 2,
      },
      Object {
        "line": 57,
        "ruleId": "no-new-func",
        "severity": 2,
      },
      Object {
        "line": 65,
        "ruleId": "no-unused-vars",
        "severity": 2,
      },
      Object {
        "line": 69,
        "ruleId": "no-implied-eval",
        "severity": 2,
      },
      Object {
        "line": 80,
        "ruleId": "no-caller",
        "severity": 2,
      },
      Object {
        "line": 83,
        "ruleId": "no-restricted-syntax",
        "severity": 2,
      },
      Object {
        "line": 83,
        "ruleId": "no-with",
        "severity": 2,
      },
      Object {
        "line": 94,
        "ruleId": "no-octal",
        "severity": 2,
      },
      Object {
        "line": 96,
        "ruleId": "no-restricted-globals",
        "severity": 2,
      },
      Object {
        "line": 97,
        "ruleId": "no-restricted-globals",
        "severity": 2,
      },
      Object {
        "line": 99,
        "ruleId": "no-restricted-properties",
        "severity": 2,
      },
      Object {
        "line": 103,
        "ruleId": "no-restricted-properties",
        "severity": 2,
      },
      Object {
        "line": 104,
        "ruleId": "no-restricted-properties",
        "severity": 2,
      },
      Object {
        "line": 105,
        "ruleId": "no-restricted-properties",
        "severity": 2,
      },
      Object {
        "line": 107,
        "ruleId": "no-restricted-properties",
        "severity": 2,
      },
      Object {
        "line": 108,
        "ruleId": "no-restricted-properties",
        "severity": 2,
      },
      Object {
        "line": 117,
        "ruleId": "eqeqeq",
        "severity": 2,
      },
      Object {
        "line": 122,
        "ruleId": "strict",
        "severity": 2,
      },
      Object {
        "line": 129,
        "ruleId": "indent",
        "severity": 2,
      },
      Object {
        "line": 129,
        "ruleId": "no-tabs",
        "severity": 1,
      },
      Object {
        "line": 131,
        "ruleId": "indent",
        "severity": 2,
      },
      Object {
        "line": 131,
        "ruleId": "no-tabs",
        "severity": 1,
      },
      Object {
        "line": 132,
        "ruleId": "no-mixed-spaces-and-tabs",
        "severity": 2,
      },
      Object {
        "line": 132,
        "ruleId": "no-tabs",
        "severity": 1,
      },
      Object {
        "line": 133,
        "ruleId": "no-mixed-spaces-and-tabs",
        "severity": 2,
      },
      Object {
        "line": 133,
        "ruleId": "no-tabs",
        "severity": 1,
      },
      Object {
        "line": 134,
        "ruleId": "no-mixed-spaces-and-tabs",
        "severity": 2,
      },
      Object {
        "line": 134,
        "ruleId": "no-tabs",
        "severity": 1,
      },
      Object {
        "line": 135,
        "ruleId": "indent",
        "severity": 2,
      },
      Object {
        "line": 135,
        "ruleId": "no-tabs",
        "severity": 1,
      },
      Object {
        "line": 137,
        "ruleId": "indent",
        "severity": 2,
      },
      Object {
        "line": 137,
        "ruleId": "no-tabs",
        "severity": 1,
      },
      Object {
        "line": 146,
        "ruleId": "one-var",
        "severity": 2,
      },
      Object {
        "line": 146,
        "ruleId": "comma-spacing",
        "severity": 2,
      },
      Object {
        "line": 146,
        "ruleId": "comma-spacing",
        "severity": 2,
      },
      Object {
        "line": 149,
        "ruleId": "key-spacing",
        "severity": 2,
      },
      Object {
        "line": 150,
        "ruleId": "key-spacing",
        "severity": 2,
      },
      Object {
        "line": 158,
        "ruleId": "key-spacing",
        "severity": 2,
      },
      Object {
        "line": 159,
        "ruleId": "key-spacing",
        "severity": 2,
      },
      Object {
        "line": 163,
        "ruleId": "switch-colon-spacing",
        "severity": 2,
      },
      Object {
        "line": 166,
        "ruleId": "switch-colon-spacing",
        "severity": 2,
      },
      Object {
        "line": 169,
        "ruleId": "switch-colon-spacing",
        "severity": 2,
      },
      Object {
        "line": 173,
        "ruleId": "space-infix-ops",
        "severity": 2,
      },
      Object {
        "line": 173,
        "ruleId": "space-infix-ops",
        "severity": 2,
      },
      Object {
        "line": 173,
        "ruleId": "space-infix-ops",
        "severity": 2,
      },
      Object {
        "line": 175,
        "ruleId": "curly",
        "severity": 2,
      },
      Object {
        "line": 176,
        "ruleId": "nonblock-statement-body-position",
        "severity": 2,
      },
      Object {
        "line": 178,
        "ruleId": "space-unary-ops",
        "severity": 2,
      },
      Object {
        "line": 180,
        "ruleId": "space-unary-ops",
        "severity": 2,
      },
    ],
  },
]
`;
